src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts

```diff
@@ -170,4 +171,49 @@ export class StorageBucketResolverMutations {
     };
     return await this.referenceService.updateReference(updateData);
   }
+
+  @UseGuards(GraphqlGuard)
+  @Mutation(() => String, {
+    description:
+      'Create a new Document on the Storage and return the value as part of the returned Reference.',
+  })
+  @Profiling.api
+  async uploadFileOnStorageBucket(
+    @CurrentUser() agentInfo: AgentInfo,
+    @Args('uploadData') uploadData: StorageBucketUploadFileInput,
+    @Args({ name: 'file', type: () => GraphQLUpload })
+    { createReadStream, filename, mimetype }: FileUpload
+  ): Promise<string> {
+    const storageBucket =
+      await this.storageBucketService.getStorageBucketOrFail(
+        uploadData.storageBucketId
+      );
+
+    this.authorizationService.grantAccessOrFail(
+      agentInfo,
+      storageBucket.authorization,
+      AuthorizationPrivilege.READ, //FILE_UPLOAD,
```

Comment:
This has to be addressed on policy level I think with the proper privilege.

Reply:
Agreed, but needs a bit of discussion about how to assign that privilege as do now want that the privilege shows up everywhere in the tree as it is very specific to the storage bucket entity. Suggest we chat it through...

One option would be to assign it in the following location:
![image](https://github.com/alkem-io/server/assets/30729240/6a06ab1f-6d69-45cd-917c-30696ef6f108)

+ then raise another issue for fixing that the FILE_UPLOAD is applied too widely. So as to not hold this issue up.

Reply:
Resolved with https://github.com/alkem-io/server/pull/2956